test/features/transaction/transaction_details_page_test.dart

```diff
@@ -0,0 +1,149 @@
+                MockTransactionNotifierWithData() => switch (
+                      mockTransactionNotifierType.transactionType) {
+                    TransactionType.send => mockSendTransactionNotifier,
+                    TransactionType.deposit => mockDepositTransactionNotifier,
+                    TransactionType.withdraw => mockWithdrawTransactionNotifier,
+                  },
+                MockTransactionNotifierWithNullData() =>
+                  nullMockTransactionNotifier,
+                MockTransactionNotifierWithError() =>
+                  erroringMockTransactionNotifier,
+              },
+            ),
+          ],
+        );
+
+    testWidgets('should show correct payout and payin amounts', (tester) async {
+      await tester.pumpWidget(
+        transactionDetailsTestWidget(
+          mockTransactionNotifierType:
+              mockTransactionNotifierWithSendTransaction,
+        ),
+      );
+      await tester.pumpAndSettle();
+
+      expect(
+        find.text('100.01'),
+        findsOneWidget,
+      ); // Adjust to your currency setup.
+    });
+
+    testWidgets('should show transaction date', (tester) async {
+      await tester.pumpWidget(
+        transactionDetailsTestWidget(
+          mockTransactionNotifierType:
+              mockTransactionNotifierWithSendTransaction,
+        ),
+      );
+      await tester.pumpAndSettle();
+
+      expect(
+        find.text(
+          DateFormat("MMM dd 'at' hh:mm a")
+              .format(sendTransaction.createdAt.toLocal()),
+        ),
+        findsOneWidget,
+      );
+    });
+
+    testWidgets('should show transaction status chip', (tester) async {
+      await tester.pumpWidget(
+        transactionDetailsTestWidget(
+          mockTransactionNotifierType:
+              mockTransactionNotifierWithSendTransaction,
+        ),
+      );
+      await tester.pumpAndSettle();
+
+      expect(
+        find.text('Order submitted'),
+        findsOneWidget,
+      ); // Adjust based on status
+    });
+
+    testWidgets('should display error when transaction fetch fails',
+        (tester) async {
+      await tester.pumpWidget(
+        transactionDetailsTestWidget(
+          mockTransactionNotifierType: mockTransactionNotifierWithError,
+        ),
+      );
+      await tester.pumpAndSettle();
+
+      expect(find.text('Bad state: Error loading transaction'), findsOneWidget);
+    });
+  });
+}
+
+sealed class MockTransactionNotifierType {
+  const MockTransactionNotifierType();
+}
+
+class MockTransactionNotifierWithData extends MockTransactionNotifierType {
+  const MockTransactionNotifierWithData({required this.transactionType});
+
+  final TransactionType transactionType;
+}
+
+class MockTransactionNotifierWithNullData extends MockTransactionNotifierType {
+  const MockTransactionNotifierWithNullData();
+}
+
+class MockTransactionNotifierWithError extends MockTransactionNotifierType {
+  const MockTransactionNotifierWithError();
+}
```

Comment:
this code is duplicated from the `TransactionTile` tests, let's move this to `mocks.dart`!